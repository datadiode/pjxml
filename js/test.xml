<?xml version="1.0" encoding="UTF-8"?>
<!-- test document for pjXML library -->
<!DOCTYPE html
     PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd" [
      <!ENTITY nbsp "&#160;">
      <!ENTITY % draft 'INCLUDE' >
      <!ENTITY % final 'IGNORE' >

      <![%draft;[
      <!ELEMENT book (comments*, title, body, supplements?)>
      ]]>
      <![%final;[
      <!ELEMENT book (title, body, supplements?)>
      ]]>
    ]>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
  <head>
    <title>Minimal XHTML 1.0 Document with W3C DTD</title>
  </head>
  <body>
    <p title="This is a&nbsp;&lt;p&gt; tag.">This is a minimal <a href="http://www.w3.org/TR/xhtml1/">XHTML 1.0</a>
    document with a W3C url for the DTD.</p>
  </body>
</html>
