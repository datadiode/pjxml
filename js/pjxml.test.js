import {expect, test} from "bun:test";
const fs = require('fs');
const pjxml = require('./pjxml.js');

test("parse()", () => {
	const xml = fs.readFileSync('test.xml', 'utf-8');
	const doc = pjxml.parse(xml);
	expect(doc).toMatchSnapshot();
	expect(doc.xml()).toMatchSnapshot();
	expect(doc.clone().xml()).toBe(doc.xml());
	const a = doc.select('//a');
	const pa = doc.pathTo(a);
	expect(pa).toMatchSnapshot();
});
