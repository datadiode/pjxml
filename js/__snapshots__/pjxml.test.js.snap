// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`parse() 1`] = `
Node {
  "attributes": undefined,
  "content": [
    "",
    Node {
      "attributes": undefined,
      "content": [
        "xml version="1.0" encoding="UTF-8"",
      ],
      "name": undefined,
      "type": 7,
    },
    
"
"
,
    Node {
      "attributes": undefined,
      "content": [
        " test document for pjXML library ",
      ],
      "name": undefined,
      "type": 8,
    },
    
"

"
,
    Node {
      "attributes": {
        "lang": "en",
        "xml:lang": "en",
        "xmlns": "http://www.w3.org/1999/xhtml",
      },
      "content": [
        
"
  "
,
        Node {
          "attributes": {},
          "content": [
            
"
    "
,
            Node {
              "attributes": {},
              "content": [
                "Minimal XHTML 1.0 Document with W3C DTD",
              ],
              "name": "title",
              "type": 1,
            },
            
"
  "
,
          ],
          "name": "head",
          "type": 1,
        },
        
"
  "
,
        Node {
          "attributes": {},
          "content": [
            
"
    "
,
            Node {
              "attributes": {
                "title": "This is a <p> tag.",
              },
              "content": [
                "This is a minimal ",
                Node {
                  "attributes": {
                    "href": "http://www.w3.org/TR/xhtml1/",
                  },
                  "content": [
                    "XHTML 1.0",
                  ],
                  "name": "a",
                  "type": 1,
                },
                
"
    document with a W3C url for the DTD."
,
              ],
              "name": "p",
              "type": 1,
            },
            
"
    "
,
            Node {
              "attributes": {},
              "content": [
                "This is a second paragraph, which contains some ",
                Node {
                  "attributes": {},
                  "content": [
                    "emphasized",
                  ],
                  "name": "em",
                  "type": 1,
                },
                " text for no reason.",
              ],
              "name": "p",
              "type": 1,
            },
            
"
  "
,
          ],
          "name": "body",
          "type": 1,
        },
        
"
"
,
      ],
      "name": "html",
      "type": 1,
    },
    
"
"
,
  ],
  "name": undefined,
  "type": 9,
}
`;

exports[`parse() 2`] = `
"


<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
  <head>
    <title>Minimal XHTML 1.0 Document with W3C DTD</title>
  </head>
  <body>
    <p title="This is a &lt;p&gt; tag.">This is a minimal <a href="http://www.w3.org/TR/xhtml1/">XHTML 1.0</a>
    document with a W3C url for the DTD.</p>
    <p>This is a second paragraph, which contains some <em>emphasized</em> text for no reason.</p>
  </body>
</html>
"
`;

exports[`parse() 3`] = `
[
  Node {
    "attributes": undefined,
    "content": [
      "",
      Node {
        "attributes": undefined,
        "content": [
          "xml version="1.0" encoding="UTF-8"",
        ],
        "name": undefined,
        "type": 7,
      },
      
"
"
,
      Node {
        "attributes": undefined,
        "content": [
          " test document for pjXML library ",
        ],
        "name": undefined,
        "type": 8,
      },
      
"

"
,
      Node {
        "attributes": {
          "lang": "en",
          "xml:lang": "en",
          "xmlns": "http://www.w3.org/1999/xhtml",
        },
        "content": [
          
"
  "
,
          Node {
            "attributes": {},
            "content": [
              
"
    "
,
              Node {
                "attributes": {},
                "content": [
                  "Minimal XHTML 1.0 Document with W3C DTD",
                ],
                "name": "title",
                "type": 1,
              },
              
"
  "
,
            ],
            "name": "head",
            "type": 1,
          },
          
"
  "
,
          Node {
            "attributes": {},
            "content": [
              
"
    "
,
              Node {
                "attributes": {
                  "title": "This is a <p> tag.",
                },
                "content": [
                  "This is a minimal ",
                  Node {
                    "attributes": {
                      "href": "http://www.w3.org/TR/xhtml1/",
                    },
                    "content": [
                      "XHTML 1.0",
                    ],
                    "name": "a",
                    "type": 1,
                  },
                  
"
    document with a W3C url for the DTD."
,
                ],
                "name": "p",
                "type": 1,
              },
              
"
    "
,
              Node {
                "attributes": {},
                "content": [
                  "This is a second paragraph, which contains some ",
                  Node {
                    "attributes": {},
                    "content": [
                      "emphasized",
                    ],
                    "name": "em",
                    "type": 1,
                  },
                  " text for no reason.",
                ],
                "name": "p",
                "type": 1,
              },
              
"
  "
,
            ],
            "name": "body",
            "type": 1,
          },
          
"
"
,
        ],
        "name": "html",
        "type": 1,
      },
      
"
"
,
    ],
    "name": undefined,
    "type": 9,
  },
  Node {
    "attributes": {
      "lang": "en",
      "xml:lang": "en",
      "xmlns": "http://www.w3.org/1999/xhtml",
    },
    "content": [
      
"
  "
,
      Node {
        "attributes": {},
        "content": [
          
"
    "
,
          Node {
            "attributes": {},
            "content": [
              "Minimal XHTML 1.0 Document with W3C DTD",
            ],
            "name": "title",
            "type": 1,
          },
          
"
  "
,
        ],
        "name": "head",
        "type": 1,
      },
      
"
  "
,
      Node {
        "attributes": {},
        "content": [
          
"
    "
,
          Node {
            "attributes": {
              "title": "This is a <p> tag.",
            },
            "content": [
              "This is a minimal ",
              Node {
                "attributes": {
                  "href": "http://www.w3.org/TR/xhtml1/",
                },
                "content": [
                  "XHTML 1.0",
                ],
                "name": "a",
                "type": 1,
              },
              
"
    document with a W3C url for the DTD."
,
            ],
            "name": "p",
            "type": 1,
          },
          
"
    "
,
          Node {
            "attributes": {},
            "content": [
              "This is a second paragraph, which contains some ",
              Node {
                "attributes": {},
                "content": [
                  "emphasized",
                ],
                "name": "em",
                "type": 1,
              },
              " text for no reason.",
            ],
            "name": "p",
            "type": 1,
          },
          
"
  "
,
        ],
        "name": "body",
        "type": 1,
      },
      
"
"
,
    ],
    "name": "html",
    "type": 1,
  },
  Node {
    "attributes": {},
    "content": [
      
"
    "
,
      Node {
        "attributes": {
          "title": "This is a <p> tag.",
        },
        "content": [
          "This is a minimal ",
          Node {
            "attributes": {
              "href": "http://www.w3.org/TR/xhtml1/",
            },
            "content": [
              "XHTML 1.0",
            ],
            "name": "a",
            "type": 1,
          },
          
"
    document with a W3C url for the DTD."
,
        ],
        "name": "p",
        "type": 1,
      },
      
"
    "
,
      Node {
        "attributes": {},
        "content": [
          "This is a second paragraph, which contains some ",
          Node {
            "attributes": {},
            "content": [
              "emphasized",
            ],
            "name": "em",
            "type": 1,
          },
          " text for no reason.",
        ],
        "name": "p",
        "type": 1,
      },
      
"
  "
,
    ],
    "name": "body",
    "type": 1,
  },
  Node {
    "attributes": {
      "title": "This is a <p> tag.",
    },
    "content": [
      "This is a minimal ",
      Node {
        "attributes": {
          "href": "http://www.w3.org/TR/xhtml1/",
        },
        "content": [
          "XHTML 1.0",
        ],
        "name": "a",
        "type": 1,
      },
      
"
    document with a W3C url for the DTD."
,
    ],
    "name": "p",
    "type": 1,
  },
]
`;
